core/templates/pages/exploration-editor-page/improvements-tab/services/improvement-suggestion-thread-modal.controller.spec.ts

```diff
@@ -0,0 +1,207 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for ImprovementSuggestionThreadModalController.
+ */
+
+describe('Improvement Suggestion Thread Modal Controller', function() {
+  var $scope = null;
+  var $uibModalInstance = null;
+  var $httpBackend = null;
+  var $q;
+  var expId = 'exp1';
+  var AlertsService = null;
+  var ThreadDataService = null;
+  var CsrfService = null;
+  var SuggestionModalForExplorationEditorService = null;
+  var thread = null;
+
+  beforeEach(angular.mock.module('oppia'));
+  beforeEach(angular.mock.module(function($provide) {
+    $provide.value('ExplorationDataService', {
+      explorationId: expId
+    });
+    $provide.value('UserService', {
+      getUserInfoAsync: function() {
+        return $q.resolve({
+          isLoggedIn: function() {
+            return true;
+          }
+        });
+      }
+    });
+    $provide.value('ExplorationStatesService', {
+      hasState: function() {
+        return true;
+      },
+      getState: function(stateName) {}
+    });
+  }));
+  beforeEach(angular.mock.inject(function($injector, $controller) {
+    AlertsService = $injector.get('AlertsService');
+    ThreadDataService = $injector.get('ThreadDataService');
+    CsrfService = $injector.get('CsrfTokenService');
+    SuggestionModalForExplorationEditorService = $injector.get(
+      'SuggestionModalForExplorationEditorService');
+    $httpBackend = $injector.get('$httpBackend');
+    $q = $injector.get('$q');
+
+    spyOn(CsrfService, 'getTokenAsync').and.callFake(function() {
```

Comment:
Same, use `.and.returnValue($q.resolve('sample-csrf-token'))`

Reply:
Done.